cmd/api.go

```diff
@@ -286,6 +292,123 @@ var conv = &cobra.Command{
+			ArgValidatorFCTAmount), // Limit is converted into a decimal object, using ArgValidatorFCTAmount for validation
+	),
+	Run: func(cmd *cobra.Command, args []string) {
+		var err error
+		cl := node.FactomClientFromConfig(viper.GetViper())
+		payment, source, srcAsset, amt, destAsset, limitStr := args[0], args[1], args[2], args[3], args[4], args[5]
+
+		// Let's check the pXXX -> pFCT first
+		status := getStatus()
+		if (destAsset == "pFCT" || destAsset == "FCT") && uint32(status.Current) >= node.OneWaypFCTConversions {
+			cmd.PrintErrln(fmt.Sprintf("pXXX -> pFCT conversions are not allowed since block height %d. If you need to acquire pFCT, you have to burn FCT -> pFCT", node.OneWaypFCTConversions))
+			os.Exit(1)
+		}
+
+		dest, _ := cmd.Flags().GetBool("dest")
+		above, _ := cmd.Flags().GetBool("above")
+		dryrun, _ := cmd.Flags().GetBool("dryrun")
+		waiting := true
+		limit, err := decimal.NewFromString(limitStr)
+		if err != nil {
+			cmd.PrintErrln("Error parsing limit amount, must be a valid number or decimal")
+			os.Exit(1)
+		}
+		var limitTicker string
+		if dest {
+			limitTicker = destAsset
+		} else {
+			limitTicker = srcAsset
+		}
+		oldRate := decimal.NewFromInt(0)
+		for waiting {
+			rates := CurrentRates()
+			newRate, _ := decimal.NewFromString(rates[limitTicker])
+			if newRate != oldRate {
+				fmt.Printf("Current Rate: %s\n", newRate)
+				if above {
+					waiting = limit.GreaterThan(newRate)
+				} else {
+					waiting = limit.LessThan(newRate)
+				}
+			} else {
+				time.Sleep(300 * time.Second)
+			}
+		}
+
+		if dryrun {
+			fmt.Println("Limit condition met, dryrun active, not converting assets")
+			return
+		}
+
+		fmt.Println("Limit condition met, converting")
+
+		// Build the transaction from the args
+		var trans fat2.Transaction
+		if err := setTransactionInput(&trans, cl, source, srcAsset, amt); err != nil {
+			cmd.PrintErrln(err.Error())
+			os.Exit(1)
+		}
+
+		if trans.Conversion, err = ticker(destAsset); err != nil {
+			cmd.PrintErrln("invalid ticker type")
+			os.Exit(1)
+		}
+
+		err, commit, reveal := signAndSend(&trans, cl, payment)
+		if err != nil {
+			cmd.PrintErrln(err.Error())
+			os.Exit(1)
+		}
+
+		fmt.Printf("conversion sent:\n")
+		fmt.Printf("\t%10s: %s\n", "EntryHash", reveal)
+		fmt.Printf("\t%10s: %s\n", "Commit", commit)
+	},
+}
+
+// CurrentRates returns the rates as of the current nodes syncheight
+func CurrentRates() map[string]string {
+	client := srv.NewClient()
+	client.PegnetdServer = viper.GetString(config.Pegnetd)
+	var res srv.ResultPegnetTickerMap
+	status := getStatus()
+	uH := uint32(status.Sync)
+	err := client.Request("get-pegnet-rates", srv.ParamsGetPegnetRates{Height: &uH}, &res)
+	if err != nil {
+		fmt.Printf("Failed to make RPC request\nDetails:\n%v\n", err)
+		os.Exit(1)
+	}
+	// Change the units to be human readable
+	humanBals := make(map[string]string)
+	for k, bal := range res {
+		humanBals[k.String()] = FactoshiToFactoid(int64(bal))
```

Comment:
In concordance with the other comment, this really isn't necessary. You're taking a uint64, converting it to a string, then using your library to convert the string to a decimal. Seems much easier to just operate on uint64 directly. You can also convert user input to ticker via `cmd.ticker()`